File: 03_Basic_Recursion/07_Reverse_Array_1Var.cs
// using recursion

int[] arr = {4, 2, 6, 8, 1, 9, 0, 7, 0};

f(arr, 0);

foreach(int element in arr)
{
    Console.Write($"{element} ");
}

void f(int[] a, int i)
{
    if (i >= a.Length / 2) return;

    (a[i], a[a.Length - i - 1]) = (a[a.Length - i - 1], a[i]);

    f(a, i + 1);
}

File: 03_Basic_Recursion/07_Reverse_Array_1Var.cpp
#include "../dsa.hpp"
using namespace std;

void f(vector<int>& a, int i)
{
    if (i >= a.size() / 2)
        return;

    swap(a[i], a[a.size() - i - 1]);

    f(a, i + 1);
}

int main()
{
    vector<int> arr = {4, 2, 6, 8, 1, 9, 0, 7, 0};

    f(arr, 0);

    for (int element : arr)
    {
        cout << element << " ";
    }

    return 0;
}